### ISSUE TITLE
A bunch of false positives

### ISSUE THREAD

**Issue by jkowalski** [esimonov/ifshort#8]
I'm using this via golangci-lint v1.37.1:

All the code can be found in https://github.com/kopia/kopia/commit/113670e57693a639336a02cfb24a0f166c00c3a7

I'm listing 4 cases here but there a few more there - I'd appreciate if you could take a look.

### Case 1

```
internal/parallelwork/parallel_work_queue.go:127:2: variable 'cb' is only used in the if-statement (internal/parallelwork/parallel_work_queue.go:128:2); consider using short syntax (ifshort)
	cb := v.ProgressCallback
	^
```

Notice how `cb` is later invoked:

```
func (v *Queue) maybeReportProgress(ctx context.Context) {
	cb := v.ProgressCallback
	if cb == nil {
		return
	}

	if clock.Now().Before(v.nextReportTime) {
		return
	}

	v.nextReportTime = clock.Now().Add(1 * time.Second)

	cb(ctx, v.enqueuedWork, v.activeWorkerCount, v.completedWork)
}
```

### Case 2

```
fs/entry.go:141:2: variable 'name' is only used in the if-statement (fs/entry.go:151:2); consider using short syntax (ifshort)
	name := newEntry.Name()

```

Notice how `name` is used in a closure passed to `sort.Search()` and in one more if statement:

```
func (e Entries) Update(newEntry Entry) Entries {
	name := newEntry.Name()
	pos := sort.Search(len(e), func(i int) bool {
		return e[i].Name() >= name
	})

	// append at the end
	if pos >= len(e) {
		return append(append(Entries(nil), e...), newEntry)
	}

	if e[pos].Name() == name {
		if pos > 0 {
			return append(append(append(Entries(nil), e[0:pos]...), newEntry), e[pos+1:]...)
		}

		return append(append(Entries(nil), newEntry), e[pos+1:]...)
	}

	if pos > 0 {
		return append(append(append(Entries(nil), e[0:pos]...), newEntry), e[pos:]...)
	}

	return append(append(Entries(nil), newEntry), e[pos:]...)
}
```

### Case 3

```
tests/testenv/cli_test_env.go:73:2: variable 'exe' is only used in the if-statement (tests/testenv/cli_test_env.go:74:2); consider using short syntax (ifshort)
	exe := os.Getenv("KOPIA_EXE")
```

Notice how `exe` is used in the return statement:

```
// NewCLITest creates a new instance of *CLITest.
func NewCLITest(t *testing.T) *CLITest {
	t.Helper()

	exe := os.Getenv("KOPIA_EXE")
	if exe == "" {
		// exe = "kopia"
		t.Skip()
	}

        /* a bunch of lines removed */

	return &CLITest{
		startTime:                    clock.Now(),
		RepoDir:                      testutil.TempDirectory(t),
		ConfigDir:                    configDir,
		Exe:                          filepath.FromSlash(exe),
		fixedArgs:                    fixedArgs,
		DefaultRepositoryCreateFlags: formatFlags,
		LogsDir:                      logsDir,
		Environment: []string{
			"KOPIA_PASSWORD=" + TestRepoPassword,
			"KOPIA_ADVANCED_COMMANDS=enabled",
		},
	}
}
```

### Case 4

```
cli/command_diff.go:35:5: variable 'isDir1' is only used in the if-statement (cli/command_diff.go:38:2); consider using short syntax (ifshort)
	_, isDir1 := ent1.(fs.Directory)
	   ^
cli/command_diff.go:36:5: variable 'isDir2' is only used in the if-statement (cli/command_diff.go:38:2); consider using short syntax (ifshort)
	_, isDir2 := ent2.(fs.Directory)
	   ^
```

Notice how isDir1 is used close to the end of the function:

```
func runDiffCommand(ctx context.Context, rep repo.Repository) error {
	ent1, err := snapshotfs.FilesystemEntryFromIDWithPath(ctx, rep, *diffFirstObjectPath, false)
	if err != nil {
		return errors.Wrapf(err, "error getting filesystem entry for %v", *diffFirstObjectPath)
	}

	ent2, err := snapshotfs.FilesystemEntryFromIDWithPath(ctx, rep, *diffSecondObjectPath, false)
	if err != nil {
		return errors.Wrapf(err, "error getting filesystem entry for %v", *diffSecondObjectPath)
	}

	_, isDir1 := ent1.(fs.Directory)
	_, isDir2 := ent2.(fs.Directory)

	if isDir1 != isDir2 {
		return errors.New("arguments do diff must both be directories or both non-directories")
	}

	d, err := diff.NewComparer(os.Stdout)
	if err != nil {
		return errors.Wrap(err, "error creating comparer")
	}
	defer d.Close() //nolint:errcheck

	if *diffCompareFiles {
		parts := strings.Split(*diffCommandCommand, " ")
		d.DiffCommand = parts[0]
		d.DiffArguments = parts[1:]
	}

	if isDir1 {
		return d.Compare(ctx, ent1, ent2)
	}

	return errors.New("comparing files not implemented yet")
}
```

**Comment by esimonov** [comment]
Thank you for such a comprehensive report. 
Cases 1 and 3 should be fixed by https://github.com/esimonov/ifshort/commit/67b69237e94c0176c5065e0c51b85e6d162589cc and https://github.com/esimonov/ifshort/commit/70b30c67a74e9156b243e588423e8ef4e62dcd19; I'll have a closer look into the other two shortly.

**Comment by esimonov** [comment]
I couldn't reproduce cases 2 and 4. I'm pretty sure they were fixed somewhere in `v1.0.2`. Please try linting with `v1.0.2` (or wait for a new release of golangci-lint if that's the way you use `ifshort`). Feel free to reopen if you still encounter those false positives.
